### PR TITLE
DBZ-960 Run SnapshotReader in a single transaction

### PR DESCRIPTION
**Pull request by jpechane**
https://issues.jboss.org/browse/DBZ-960

### CODE REVIEW

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java [hunk 1]
```diff
@@ -620,6 +620,7 @@ protected void execute() {
                 }
                 step++;
             } finally {
+                mysql.connection().commit();
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java]: This looks odd here, given the rollback attempt below (and second `commit()` in the else branch). Can this just be removed?

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlJdbcContext.java [hunk 2]
```diff
@@ -258,7 +258,8 @@ protected String connectionString() {
 
     private Map<String, String> querySystemVariables(String statement) {
         Map<String, String> variables = new HashMap<>();
-        try (JdbcConnection mysql = jdbc.connect()) {
+        try {
+            JdbcConnection mysql = jdbc.connect();
             mysql.query(statement, rs -> {
```
Comment (gunnarmorling) [inline debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlJdbcContext.java]: It seems the line before could just be removed and here it'd be `jdbc.query(...)`?

Comment (jpechane) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlJdbcContext.java]: done

File: debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java (hunk 1 above)
Comment (jpechane) [reply debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java]: I've changed the `commit` to `rollback` and placed it to a separata branch to cover all potenital cases.

**Comment by gunnarmorling** [comment]
Merged; thanks, @jpechane!

**Pull request merged**